state/factory/workingset.go

```diff
@@ -168,7 +168,13 @@ func withActionCtx(ctx context.Context, selp *action.SealedEnvelope) (context.Co
 func (ws *workingSet) runAction(
 	ctx context.Context,
 	selp *action.SealedEnvelope,
-) (*action.Receipt, error) {
+) (receipt *action.Receipt, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			receipt = nil
+			err = errors.Wrapf(action.ErrPanic, "panic when running action: %v", r)
```

Comment:
this is an error which should be caught and handled appropriately in `pickAndRunActions` such that the block mint process can continue

Reply:
If we want to continue minting block, we need to revert to the state after the previous transaction is executed. This may require calling `Snapshot()` before transaction execution, which would likely result in a hard fork.